fission-cli/library/Fission/CLI/Handler/User/Register.hs

```diff
@@ -56,12 +53,12 @@ register ::
   )
   => Maybe Username
   -> Maybe Email
-  -> m (Either (OpenUnion Errs) Username)
+  -> m Username
 register maybeUsername maybeEmail = do
   attempt (sendAuthedRequest User.whoami) >>= \case
     Right username -> do
       CLI.Success.alreadyLoggedInAs $ textDisplay username
-      Just username
```

Comment:
Putting the `Just` here at this line specialized the entire contextual `m` to be `Maybe`. The type inference said "oh, you're return is a `Maybe`! So then the type is `Maybe (Either (OpenUnion Errs) Username)`, right?!".

```haskell
data Maybe a 
  = Just a 
  | Nothing -- Error case
```

`Maybe` can't carry the `ClientError`, because it has nowhere to put it in that `Nothing`. Because the only way to raise things with a `Maybe` throws away information, it can only raise `()` (showing that it's super lossy).